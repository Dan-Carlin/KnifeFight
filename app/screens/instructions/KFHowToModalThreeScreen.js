/*
KFHowToModalThreeScreen - View for third screen of instructions modal.
*/

// External libraries
import React from "react";
import { View } from "react-native";

// Assets
import {
  ArrowLeftGraphic,
  ArrowRightGraphic,
} from "../../assets/buttons/actions";
import { DiagramCGraphic } from "../../assets/diagrams";

// Components
import OpacityButton from "../../components/OpacityButton";
import Text from "../../components/Text";

// Resources
import sounds from "../../assets/sounds/sounds";
import strings from "../../config/strings";
import styles from "./KFHowToModalThreeStyles";

function KFHowToModalThreeScreen({ onPreviousPress }) {
  return (
    <View style={styles.screenContainer}>
      <View style={styles.bodyContainer}>
        <View style={styles.instructionsContainer}>
          <View style={styles.numberContainer}>
            <Text>{strings.modal_number_three_label}</Text>
          </View>
          <View style={styles.textContainer}>
            <Text>
              <Text style={styles.subtitle}>
                {strings.modal_instructions_three_subtitle}
              </Text>
              <Text>{strings.modal_instructions_three_text_a}</Text>
              <Text style={styles.highlightText}>
                {strings.modal_instructions_three_highlight}
              </Text>
              <Text>{strings.modal_instructions_three_text_b}</Text>
            </Text>
          </View>
        </View>
        <View style={styles.imageContainer}>
          <DiagramCGraphic />
        </View>
      </View>
      <View style={styles.buttonsContainer}>
        <View style={styles.button}>
          <OpacityButton
            Graphic={ArrowLeftGraphic}
            onPress={onPreviousPress}
            sound={sounds.ARROW_LEFT}
          />
        </View>
        <View style={styles.pageLabelContainer}>
          <Text style={styles.pageLabel}>{strings.modal_page_three_label}</Text>
        </View>
        <View style={styles.button}>
          <ArrowRightGraphic style={styles.buttonDisabled} />
        </View>
      </View>
    </View>
  );
}

export default KFHowToModalThreeScreen;
