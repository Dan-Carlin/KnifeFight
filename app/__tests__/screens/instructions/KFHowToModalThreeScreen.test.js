import React from "react";
import renderer from "react-test-renderer";

import KFHowToModalThreeScreen from "../../../screens/instructions/KFHowToModalThreeScreen";

describe("How To Modal third screen tests", () => {
  it("renders third How To screen correctly", () => {
    const tree = renderer.create(<KFHowToModalThreeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
