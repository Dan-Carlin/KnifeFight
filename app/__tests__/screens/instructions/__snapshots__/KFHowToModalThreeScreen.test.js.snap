// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`How To Modal third screen tests renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.85,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "marginEnd": 6,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          3.
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            During your rival's turn: 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            The screen will automatically rotate to display the information needed by your rivals during their turns, once it's your turn again, tap the 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "color": "#61bb46",
                  "fontStyle": "italic",
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Begin Turn
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
             button to return to the tools screen.
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "padding": 10,
        }
      }
    >
      <SvgMock />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.15,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "height": "80%",
          "width": "25%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <SvgMock />
      </View>
    </View>
    <View
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "width": "100%",
            },
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        3/3
      </Text>
    </View>
    <View
      style={
        Object {
          "height": "80%",
          "width": "25%",
        }
      }
    >
      <SvgMock
        style={
          Object {
            "opacity": 0.5,
          }
        }
      />
    </View>
  </View>
</View>
`;
